src/ImageSharp/Image.FromBytes.cs

```diff
@@ -27,7 +26,7 @@ public static IImageFormat DetectFormat(ReadOnlySpan<byte> data)
     /// <param name="data">The byte span containing encoded image data to read the header from.</param>
     /// <exception cref="ArgumentNullException">The options are null.</exception>
     /// <returns>The mime type or null if none found.</returns>
-    public static IImageFormat DetectFormat(DecoderOptions options, ReadOnlySpan<byte> data)
+    public static IImageFormat? DetectFormat(DecoderOptions options, ReadOnlySpan<byte> data)
```

Comment:
We still need to convert this and above to use the Try...Out pattern

Reply:
Do you want to convert InternalDetectFormat from Image.Decode.cs to the Try Pattern or just return IImageFormat?

Reply:
Ideally it should use the Try pattern all the way down.

Reply:
Having a second look and we wouldn't be able to adopt the Try pattern there actually. The method is used within a lambda which doesn't allow ref or out parameters..